src/common/FileSystem.cpp

```diff
@@ -2155,37 +2171,77 @@ bool IsInitialized()
 // Add a pak to the list of available paks
 static void AddPak(pakType_t type, Str::StringRef filename, Str::StringRef basePath)
 {
+	size_t suffixLen;
+	bool isPakDir = (type == pakType_t::PAK_DIR);
 	std::string fullPath = Path::Build(basePath, filename);
 
-	size_t suffixLen = type == pakType_t::PAK_DIR ? strlen(PAK_DIR_EXT) : strlen(PAK_ZIP_EXT);
+	suffixLen = type == pakType_t::PAK_DIR ? strlen(PAK_DIR_EXT) : strlen(PAK_ZIP_EXT);
+
 	std::string name, version;
 	Util::optional<uint32_t> checksum;
-	if (!ParsePakName(filename.begin(), filename.end() - suffixLen, name, version, checksum) || (type == pakType_t::PAK_DIR && checksum)) {
-		if (!UseLegacyPaks()) {
-			fsLogs.Warn("Invalid pak name: %s", fullPath);
+
+	if (!ParsePakName(filename.begin(), filename.end() - suffixLen, name, version, checksum) || (isPakDir && checksum)) {
+       fsLogs.Warn("Invalid pak name: '%s'", fullPath);
+       return;
+	}
+
+	if (type == pakType_t::PAK_ZIP) {
+		if (!isdigit(version.at(0))) {
```

Comment:
I modified that part like this:

```diff
        if (type == pakType_t::PAK_ZIP) {
+               if (version.at(0) > '9') {
+                       fsLogs.Warn("Pak version is very high: '%s'", fullPath);
+               }
                fsLogs.Verbose("Found pak: '%s'...", fullPath);
```

It enforces nothing, user is free to make mistakes, but he gets a warning saying a01 (for example) is a very high version number, something that he probably not expect from an alpha version number. Can be helpful.